app/controllers/sessions_controller.rb

```diff
@@ -1,20 +1,27 @@
 class SessionsController < Clearance::SessionsController
+  include PrivacyPassSupportable
+
   before_action :redirect_to_signin, unless: :signed_in?, only: %i[verify authenticate]
   before_action :redirect_to_new_mfa, if: :mfa_required_not_yet_enabled?, only: %i[verify authenticate]
   before_action :redirect_to_settings_strong_mfa_required, if: :mfa_required_weak_level_enabled?, only: %i[verify authenticate]
   before_action :ensure_not_blocked, only: :create
 
+  def new
+    if redeem_privacy_pass_token
+      super
+    else
+      setup_privacy_pass_challenge
```

Comment:
nit: i think some of this might go better in a `before_action unless: :redeem_privacy_pass_token`